### PR TITLE
Add transform method to Pytorch container

### PR DESCRIPTION
**Pull request by interesaaat**
To be merged after #145.

### CODE REVIEW

File: hummingbird/ml/operator_converters/_base_operator.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from abc import ABC
+
+
+class BaseOperator(ABC):
+    """
+    Abstract class definig the basic structure for operator implementations in Hummingbird.
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/_base_operator.py]: s/definig/defining/g

File: hummingbird/ml/_container.py [hunk 2]
```diff
@@ -69,6 +69,17 @@ def forward(self, *inputs):
             else:
                 return list(variable_map[output_name] for output_name in self.output_names)
 
+
+class PyTorchBackendModelTransformer(PyTorchBackendModel):
+    def transform(self, *inputs):
+        """
+        Utility functions used to emulate the behavior of the Sklearn API.
+        On data transforerms it returns trandormed output data
```
Comment (ksaur) [inline hummingbird/ml/_container.py]: s/trandormed/transformed/g

File: hummingbird/ml/_container.py [hunk 3]
```diff
@@ -69,6 +69,17 @@ def forward(self, *inputs):
...
+        """
+        return self.forward(*inputs).cpu().numpy().flatten()
```
Comment (ksaur) [inline hummingbird/ml/_container.py]: I know this is correct and necessary but....oh numpy.

File: hummingbird/ml/operator_converters/_base_operator.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        # An operator can be either a model or a transformer. In the latter case, self.transformer must be set to True.
+        # In the former case, if a model is doing regression, then self.regression must be set to True. Otherwise the operator is doing some classificaiton task.
+        self.regression = False
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/_base_operator.py:14]: I think there are a lot of places that will be needed to be updated to use these, right?

File: hummingbird/ml/operator_converters/_base_operator.py (hunk 1 above)
Comment (interesaaat) [reply hummingbird/ml/operator_converters/_base_operator.py]: of course..

File: hummingbird/ml/_container.py (hunk 3 above)
Comment (interesaaat) [reply hummingbird/ml/_container.py]: Yea.

File: hummingbird/ml/operator_converters/_base_operator.py (hunk 4 above)
Comment (interesaaat) [reply hummingbird/ml/operator_converters/_base_operator.py:14]: Nope, I was already doing this in the tree classes, I just moved all implementations to subclass this.

**Comment by interesaaat** [comment]
Ready for a second pass! I think I fixed everything.

**Pull request merged**